src/index.ts

```diff
@@ -147,7 +148,7 @@ function resolveLaravelPlugin(pluginConfig: Required<PluginConfig>): LaravelPlug
             const hotFile = path.join(pluginConfig.publicDirectory, 'hot')
 
             const envDir = resolvedConfig.envDir || process.cwd()
-            const appUrl = loadEnv('', envDir, 'APP_URL').APP_URL
+            const appUrl = loadEnv(resolvedConfig.mode, envDir, 'APP_URL').APP_URL
```

Comment:
Something like this would prevent the error you're seeing:

```suggestion
            const appUrl = loadEnv(resolvedConfig.mode, envDir, 'APP_URL').APP_URL ?? 'undefined'
```

It would ensure that `appUrl` is a string when it doesn't exist in any of env files, which would prevent the `.replace()` call from failing.

The output would then look like this, which should hopefully make it obvious that the `APP_URL` couldn't be found.

![image](https://user-images.githubusercontent.com/4977161/183316342-f3fdd750-2d6d-402e-8c54-88e2c4df0030.png)

We could throw an error if no `APP_URL` exists, but it's not required for this plugin to function.

Reply:
I agree, would be a great idea. Just commited :)